server/src/routes/api/v1/reviewsRouter.js

```diff
@@ -0,0 +1,15 @@
+import express from "express";
+
+import { Review, Dog } from "../../../models/index.js";
+import getReviewsDetails from "../../../serializers/ReviewSerializer.js";
+
+const reviewsRouter = new express.Router();
+
+reviewsRouter.get("/:dogId", async (req, res) => {
+  const dog = await Dog.query().findById(req.params.dogId);
+  const reviews = await dog.$relatedQuery("reviews");
+  const serializedReviews = await getReviewsDetails(reviews);
+  res.status(200).json({ reviews: serializedReviews });
+});
```

Comment:
actually, this isnt what we want. This means two fetch requests will fire off on the ShowPage upon loading: one for the dog, one for the dog's review. We want to retrieve BOTH the dog and the reviews for that dog in the original fetch requests for the dog. You can use a serializer to link the review. That way you only have one fetch request, fewer problems with async stuff, and the page doesnt have to go through as many unnecessary page reloads

Reply:
Ahh - yep this was all ways wrong.  We now have once getDogs fetch that get the serialized reviews for the dog in one fetch.